src/pixelator/cli/layout.py

```diff
@@ -0,0 +1,107 @@
+"""
+Console script for pixelator (layout)
+
+Copyright © 2024 Pixelgen Technologies AB.
+"""
+
+from pathlib import Path
+from typing import get_args
+
+import click
+
+from pixelator import read
+from pixelator.cli.common import logger, output_option
+from pixelator.graph.backends.protocol import SupportedLayoutAlgorithm
+from pixelator.pixeldataset.precomputed_layouts import (
+    generate_precomputed_layouts_for_components,
+)
+from pixelator.report.models.layout import LayoutSampleReport
+from pixelator.utils import (
+    create_output_stage_dir,
+    get_sample_name,
+    log_step_start,
+    sanity_check_inputs,
+    timer,
+    write_parameters_file,
+)
+
+
+@click.command(
+    "layout",
+    short_help="compute graph layouts that can be used to visualize components",
+    options_metavar="<options>",
+)
+@click.argument(
+    "pxl_file",
+    nargs=1,
+    required=True,
+    type=click.Path(exists=True),
+    metavar="PXL",
+)
+@click.option(
+    "--add-node-marker-counts",
+    required=False,
+    is_flag=True,
+    default=True,
+    help="Add marker counts to the layout. Default: True",
+)
+@click.option(
+    "--layout",
+    required=False,
+    multiple=True,
+    default=["pmds_3d"],
+    help="Select a layout algorithm to use. This can be specified multiple times to compute multiple layouts. Default: pmds_3d",
+    type=click.Choice(get_args(SupportedLayoutAlgorithm)),
+)
+@output_option
+@click.pass_context
+@timer
+def layout(
+    ctx,
+    pxl_file,
+    add_node_marker_counts,
+    layout,
+    output,
+):
+    """
+    Compute graph layouts that can be used to visualize components
+    """
+    log_step_start(
+        "layout",
+        input_files=pxl_file,
+        add_node_marker_counts=add_node_marker_counts,
+        layout=layout,
+        output=output,
+    )
+
+    # some basic sanity check on the input files
+    sanity_check_inputs(pxl_file, allowed_extensions="pxl")
+
+    # create output folder if it does not exist
+    output_path = Path(output)
+    layout_output_dir = create_output_stage_dir(output_path, "layout")
```

Comment:
I think we should phase this utility function out and replace it with PixelatorWorkdir.stage_dir

```suggestion
    workdir = PixelatorWorkdir(output)
    layout_output_dir = workdir.stage_dir("layout")
```